Rename SuccessURLAllowedHostsMixin to RedirectURLMixin and hint missing attributes

si14 opened the pull request:
Just a simple stub for `contrib.auth.views.RedirectURLMixin`.

I noticed that it was in the todolist and I needed it, so here it is.

si14 committed:
add a stub for RedirectURLMixin

intgr commented on django-stubs/contrib/auth/views.pyi:
```diff
@@ -9,6 +9,15 @@ from django.views.generic.edit import FormView
+    def get_success_url_allowed_hosts(self) -> set[str]: ...
+    def get_default_redirect_url(self) -> str: ...
+
 class SuccessURLAllowedHostsMixin:
```
Some more changes look warranted...

`SuccessURLAllowedHostsMixin` no longer exists in Django 4.2, `LoginView` and `LogoutView` should inherit from `RedirectURLMixin` instead.

intgr reviewed (changes requested):
Please remove `SuccessURLAllowedHostsMixin` and replace it with the new `RedirectURLMixin`.

si14 committed:
remove SuccessURLAllowedHostsMixin and replace it with RedirectURLMixin

Per Django 4.1 release notes
(https://docs.djangoproject.com/en/4.2/releases/4.1/#miscellaneous):

"The undocumented django.contrib.auth.views.SuccessURLAllowedHostsMixin
mixin is replaced by RedirectURLMixin."

intgr reviewed (approved):
Thank you!
